Yona/Yona/Friends/AddFriendsViewController.swift

```diff
@@ -170,121 +156,169 @@ class AddFriendsViewController: UIViewController, UIScrollViewDelegate, UINaviga
-                                                                })
-                                                            } else {
-                                                                print("Access not granted")
-                                                            }
-                                                            
-                })
-            }
+    
+    // MARK: Touch Event of Custom Segment
+    extension AddFriendsViewController {
+        
+        @IBAction func ManualTabAction(sender: AnyObject) {
+            addressBookTabView.alpha = 0.5
+            addressBookTabBottomBorder.hidden = true
+            manualTabView.alpha = 1.0
+            manualTabBottomBorder.hidden = false
+        }
+        
+        @IBAction func AddressBookTabAction(sender: AnyObject) {
+            manualTabView.alpha = 0.5
+            manualTabBottomBorder.hidden = true
+            addressBookTabView.alpha = 1.0
+            addressBookTabBottomBorder.hidden = false
             
-        case .Restricted:
-            print("Access restricted")
+            addressbookAccess()
             
-        default:
-            print("Other Problem")
         }
     }
-}
-
-
-// MARK: Touch Event of Custom Segment
-extension AddFriendsViewController {
     
-    @IBAction func ManualTabAction(sender: AnyObject) {
-        addressBookTabView.alpha = 0.5
-        addressBookTabBottomBorder.hidden = true
-        manualTabView.alpha = 1.0
-        manualTabBottomBorder.hidden = false
-    }
     
-    @IBAction func AddressBookTabAction(sender: AnyObject) {
-        manualTabView.alpha = 0.5
-        manualTabBottomBorder.hidden = true
-        addressBookTabView.alpha = 1.0
-        addressBookTabBottomBorder.hidden = false
+    extension AddFriendsViewController: UITextFieldDelegate {
         
-        addressbookAccess()
+        func textFieldDidBeginEditing(textField: UITextField) {
+            if textField == mobileTextfield {
+                IQKeyboardManager.sharedManager().enableAutoToolbar = true
+            } else {
+                IQKeyboardManager.sharedManager().enableAutoToolbar = false
+            }
+        }
         
+        func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
+            if (textField == mobileTextfield) {
+                if ((previousRange?.location >= range.location) ) {
+                    if (textField.text?.utf16.count ?? 0) + string.utf16.count - range.length == YonaConstants.mobilePhoneSpace.mobileFirstSpace || (textField.text?.utf16.count ?? 0) + string.utf16.count - range.length == YonaConstants.mobilePhoneSpace.mobileMiddleSpace {
+                        textField.text = String(textField.text!.characters.dropLast())
+                        textField.text = String(textField.text!.characters.dropLast())
```

Comment:
I removed the repeated line not sure about the rest... @chandanvarma009 can you fix this in the morning...